Description: ignore cgroup2 under systemd
 Newer versions of systemd mount a unified view of the cgroups (cgroup2)
 in the cgroup directory.  Docker assumes everything in here is a cgroup
 type and explodes when it is not.  Ignore this unified view in the short
 term as it is not necessarily going to remain constant and has unknown
 security impact.
Author: Andy Whitcroft <apw@canonical.com>
Bug-Ubuntu: https://launchpad.net/bugs/1692912
Last-Update: 2017-05-23

--- runc-1.0.0~rc2+docker1.12.6.orig/libcontainer/rootfs_linux.go
+++ runc-1.0.0~rc2+docker1.12.6/libcontainer/rootfs_linux.go
@@ -283,6 +283,10 @@ func getCgroupMounts(m *configs.Mount) (
 	var binds []*configs.Mount
 
 	for _, mm := range mounts {
+		// the unified hierachy is not a cgroup, for now ignore it.
+		if mm.Mountpoint == "/sys/fs/cgroup/unified" {
+			continue
+		}
 		dir, err := mm.GetThisCgroupDir(cgroupPaths)
 		if err != nil {
 			return nil, err
